packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -25,6 +26,50 @@ enum Orientation {
   landscape
 }
 
+/// A specific part of MediaQueryData to depend on
+enum MediaQueryAspect {
```

Comment:
This can be private?

Reply:
If you make this private, you also wouldn't have to worry about the documentation.. :)

Reply:
I don't think it can be private since it's part of the class definition of MediaQuery:

```dart
class MediaQuery extends InheritedModel<MediaQueryAspect> {
  //...
}
```

Reply:
Private types should be allowed in the "extends" clause, AFAIK. What is complaining?

Reply:
It's not a compilation error or anything, I just remember I tried it before and got an error on one of the CI checks. Maybe the documentation website generation? Will try it again to see the exact issue.

Reply:
```
   info • Avoid using private types in public APIs • packages/flutter/lib/src/widgets/media_query.dart:1284:62 • library_private_types_in_public_api
```
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8799025370985938833/+/u/Analyze/analyze/test_stdout

Reply:
Oh, it appears it is complaining about the private usage in this method signature:

```dart
bool updateShouldNotifyDependent(MediaQuery oldWidget, Set<_MediaQueryAspect> dependencies)
```

I think, you can just type this as follows and the warning goes away and you can keep it private. (we do a similar trick for private RenderObjects in `updateRenderObject`):

```dart
bool updateShouldNotifyDependent(MediaQuery oldWidget, Set<Object> dependencies)
```